### DISCUSSION TITLE
Including a transgene in ArchR GeneScore matrix?

### DISCUSSION THREAD

**Discussion by colin986** [GreenleafLab/ArchR#1902] [Questions / Documentation]
Hi, 

First thanks for developing ArchR, it's a great package. 

In our system, I'm interested in analysing the accessibility of a transgene and are only are interested in getting an estimate of how many of our cells have retained the transgene.  

I'm getting an error when calculating the GeneScore matrix. I'm pretty sure that this is because of the extension upstream from the start of the transgene (the first coordinate becomes a negative number) which is approx. 8kb and contains two genes of interest approx 4kb from the 5' end. 

My question is, is it possible to change the parameters for calculation of the GeneScore for genes in transgene (i.e. having no extension upstream) while keeping the calculation standard for the genes in the host cell's genome?

Any suggestions/help at all would be welcome

TIA,
Colin

**Comment by rcorces** [comment]
no this is not possible. I've never had to do what you are doing but I think I would approach this by creating my own custom genome and genome annotation with the transgene inserted in the correct place.

**Reply by colin986** [reply]
Thanks for responding Ryan, 

The issue with our system is that our transgene is integrated randomly across the genome and will likely be in several places. The transgene is prone to copy number loss. Our cell line is clonal and we have one version of the cell line that produces the protein of interest from the transgene and a later passage that does not. 

We know what we are trying to do is imperfect, we just want a general sense of the accessibility for the transgene(s) in any cell. We also have scRNA-seq from the same cells.

What would you think of appending the transgene to the end of a scaffold in the genome where there is no annotated in gene within 10kb of the end, updating the GTF with the new coordinates? This would permit the extension upstream 5kb, where ideally there should be no fragments? 

Colin

**Reply by rcorces** [reply]
why not handle the transgene separately? Why bother trying to force ArchR to calculate the gene score for you? it seems like all you really want is the sum of insertion counts within the transgene?

**Reply by colin986** [reply]
I hadn't considered that approach Ryan and it seems a reasonable one to me. 

When say the sum the insertion counts, do you mean that this should be done outside ArchR from a fragment file?. I presume that the barcode could be passed from ArchR if we want to determine the counts for a particular cluster? 

Thanks for your time. 
Colin

**Reply by rcorces** [reply]
I was picturing doing this outside of ArchR by counting reads mapped to a genome made for the transgene. Then getting the number of reads per cellular barcode and matching that to the barcodes in ArchR and adding this to `cellColData`
